src/Native/Utils.js

```diff
@@ -16,7 +16,13 @@ function eq(rootX, rootY)
 		{
 			continue;
 		}
-		if (typeof x === 'object')
+
+		if (x instanceof Date && y instanceof Date)
+		{
+			if (x.valueOf() !== y.valueOf())
+			{ return false; }
```

Comment:
also for what it's worth, I think you should probably just do `return x.valueOf() === y.valueOf()` since there's no point doing anything else.